addons/website/tools.py

```diff
@@ -200,3 +200,31 @@ def get_unaccent_sql_wrapper(cr):
     if odoo.registry(cr.dbname).has_unaccent:
         return lambda x: sql.SQL("unaccent({wrapped_sql})").format(wrapped_sql=x)
     return lambda x: x
+
+
+def add_form_signature(html_fragment, env_sudo):
+    if not html_fragment.findall('.//form'):
+        # Most efficient way to discard the function if no form in the view
+        return
```

Comment:
It's not the case when coming from `_get_combined_arch()`, we still need an "efficient" short-circuit check.
Ideally, I would only have a check here but I had to introduce one in `value_to_html()` (the other caller point) to avoid useless operation if there is no `form` inside it (turning it into fragment to pass to the function).

Actually, nothing changed regarding the `_get_combined_arch()` implementation, apart from the fact the code got encapsulated to be re-used at the other call point.